packages/react/src/provider/provider.tsx

```diff
@@ -20,26 +22,18 @@ type ClientOrDomain =
 
 type ProviderProps = {
   children?: React.ReactNode;
-} & ClientOrDomain;
-
-const Context = React.createContext<Client | undefined>(undefined);
-
-export const OpenFeatureProvider = ({ client, domain, children }: ProviderProps) => {
+} & ClientOrDomain &
+  ReactFlagEvaluationOptions;
+
+  /**
+   * Provides a scope for evaluating feature flags by binding a client to all child components.
+   * @param {ProviderProps} properties props for the context provider
+   * @returns {OpenFeatureProvider} context provider
+   */
+export function OpenFeatureProvider({ client, domain, children, ...options }: ProviderProps) {
   if (!client) {
     client = OpenFeature.getClient(domain);
   }
 
-  return <Context.Provider value={client}>{children}</Context.Provider>;
-};
-
-export const useOpenFeatureClient = () => {
-  const client = React.useContext(Context);
-
-  if (!client) {
-    throw new Error(
-      'No OpenFeature client available - components using OpenFeature must be wrapped with an <OpenFeatureProvider>'
-    );
-  }
-
-  return client;
-};
+  return <Context.Provider value={{ client, options }}>{children}</Context.Provider>;
```

Comment:
I was wondering about having a default suspense boundary in the `OpenFeatureProvider`.
Thinking about it I think it is no good idea as the following

```ts
return (
  <App>
   <Suspense ...>
      <OpenFeatureProvider ... >
        <MyFeatureFlagConsumingComponent>
      </OpenFeatureProvider>
   </Suspense>
  </App>
```
 results in a higher level suspense not being triggered without e.g. disabling the default suspense from `OpenFeatureProvider`.
On the other hand we could add a suspense if a suspense component is given like:

```ts
return (
  <App>
      <OpenFeatureProvider ... onSuspense={<div>Loading...</div>} >
        <MyFeatureFlagConsumingComponent>
      </OpenFeatureProvider>
  </App>
```

In the end I think this only adds complexity to avoid folks using `Suspense` directly which is a React primitive.
Just thought a bit about it and would like to get others opinion @toddbaert.

Reply:
🤔 

I'll think about this some more.

Reply:
These are both good things to consider, but I think they take control aware from devs. I'm glad you brought them up though because they are possible alternatives to the existing API.

I think a built is suspense is not good for the reason you say: 

> higher level suspense not being triggered without e.g. disabling the default suspense from OpenFeatureProvider.

> `<OpenFeatureProvider ... onSuspense={<div>Loading...</div>} >`

This is a more compelling idea, but I think it also takes too much freedom away from users. Particularly, this would result in suspending everything inside the Context Provider, which will work for simple cases like the [react-sdk-demo](https://github.com/open-feature/react-test-app). It's not great however, when you have multiple components in a UI (for instance, multiple cards) and you want them to load independently. Some of our internal UI's using the React SDK work like this (1 or 2 smaller views in the large UI use flags and have their own suspense boundary); they load independently of other components when the provider becomes ready. I think doing "built-in" suspense as proposed above will discourage or confuse the pattern I describe, which I don't want.